### PR TITLE
Use config_checker as the healthz checker

### PR DESCRIPTION
**Pull request by ycao56**
Use config_checker as the healthz checker so that the pod can be
restarted proactively when hub kubeconfig is changed. This is
necessary for backup and restore scenario.

For issue: open-cluster-management/backlog#17651

Signed-off-by: Yu Cao <ycao@redhat.com>

### CODE REVIEW

File: tool/options.go
Comment (mprahl) [inline tool/options.go:78]: This value needs to be used in `manager.Options`.

**Review by mprahl — CHANGES_REQUESTED** [review]

File: tool/options.go
Comment (ycao56) [reply tool/options.go:78]: Thanks. I was wondering why the kind test failed at pod not ready. You saved me a lot of time!

File: main.go
Comment (mprahl) [inline main.go:110]: Ideally this should have been a separate commit since it's a separate logical change that we want to keep even if we decided in the unlikely event to revert the commit introducing the health and readiness checks.

I won't block on it though.  :slightly_smiling_face:

**Review by mprahl — APPROVED** [review]
Looks great!

/hold until you are ready to merge.

**Pull request merged**
